File: dart/lib/src/base/models/rectangle.dart
import "package:shapes/base/commons.dart";

class Rectangle implements Dimensional {
  double? _width;
  double? _height;

  Rectangle(double width, double height) {
    double dimOne = width > 0 ? width : 1;
    double dimTwo = height > 0 ? height : 1;
    _width = dimOne;
    _height = dimTwo;
  }

  @override
  double area() {
    return _width! * _height!;
  }

  @override
  double perimeter() {
    return 2 * (_width! + _height!);
  }
}

File: php/src/Base/Models/Rectangle.php
<?php

namespace Shapes\Base\Models;

use Shapes\Base\Commons\Dimensional;

final class Rectangle implements Dimensional {
    private float $width;
    private float $height;

    public function __construct(float $width, float $height) {
        $dim_one = $width > 0 ? $width : 0;
        $dim_two = $height > 0 ? $height : 0;
        $this->width = $dim_one;
        $this->height = $dim_two;
    }

    #[\Override]
    public function area(): float {
        return $this->width * $this->height;
    }

    #[\Override]
    public function perimeter(): float {
        return 2.0 * ($this->width + $this->height);
    }
}
